### ISSUE TITLE
[Bug] Upload image not working

### ISSUE THREAD

**Issue by madeus** [DifuseHQ/Kalmia#17]
* **I'm submitting a**:

   - [x ] Bug Report

  * **Version**: 0.0.2
  * **Browser**: [all]
  * **Operating System**: Roxcky Linux 9.5
  
I try to add a photo, and then I see only an error during upload. The logs are empty. SELinux is clean.

![image](https://github.com/user-attachments/assets/b164c909-6311-4b49-9412-e4aff22c2a70)

And the second question also concerns images. What should be the correct path if I want to change the logo? I've tried every possible option and none of them work.

![image](https://github.com/user-attachments/assets/23f06e9d-28ca-4626-9aae-bcb1c9de4a2d)

**Comment by hayzamjs** [comment]
Do you have a valid S3 config in your `config.json`? Also are there some logs on the server side? You could add `"logLevel": "debug",` too to the `config.json` to get some verbose messages

**Comment by madeus** [comment]
I don't have S3. Do I need to add photos?

**Comment by madeus** [comment]
Are you saying that I need to host the logos somewhere outside of the Kalmia instance itself?

**Comment by hayzamjs** [comment]
Yes, currently we use S3 for dealing with image/other file storage, and yes for the favicon/other images I mentioned (that pertains to documentation edit itself) earlier they need to be hosted somewhere atm, although it makes sense to have a file input here that can store to the db because it makes things simpler, we'll look into that.

**Comment by madeus** [comment]
Not everyone has the ability (access) to S3. It would be nice to have other (more "local") options. ;-)

**Comment by hayzamjs** [comment]
I understand your point, and having some crucial images stored in the database is a good idea. However, regarding other images, I don't think we'll add any more options, as it doesn't have to be Amazon's S3 specifically; it can be any S3-compatible storage. Earlier in issue #7, a user mentioned using MinIO, if I recall correctly. We use MinIO in-house for testing and Cloudflare R2 for production.

**Comment by madeus** [comment]
This doesn't change the fact that S3 is not available to everyone. I can use QuObjects from QNAP, for example, because I have access to it. Another has MinIO. But not everyone has such capabilities. It's worth considering other options.

Anyway, I'll check S3 (from QNAP) and let you know.

**Comment by madeus** [comment]
I added S3 configuration:

```
"s3": {
    "endpoint": "https://domain.pl:8010",
    "region": "auto",
    "accessKeyId": "************************",
    "secretAccessKey": "************************",
    "bucket": "DocS3Bucket",
    "usePathStyle": true,
    "publicUrlFormat": "https://domain.pl:8010/v1/AUTH_DocS3/DocS3Bucket/%s"
  }
```

Same error. The only difference is that when I add e.g. an image as Embed (the link leads to S3 of course), the image is visible correctly. However, when I try to add it as Upload, I get the same error.

**Comment by hayzamjs** [comment]
Unfortunately I cannot reproduce the issue (I tried image upload using both the button and copy pasting to the editor which is what I usually do), can you give me a video of the problem happening? And also browser console logs if any.

**Comment by madeus** [comment]
![kalmia_upload](https://github.com/user-attachments/assets/f2df8a4f-4aad-4db8-98f2-1ac23ecb2dac)

![image](https://github.com/user-attachments/assets/8253e12a-e665-445b-8a11-15d1e84fbacb)

**Comment by mirsahebali** [comment]
Hey @madeus, check our latest changelog of kalmia v0.2.0 1bc059450c3901b17e1be99209965b12f6d163e4, we have implemented fully local file uploading, so you can use it and host it fully locally inside a single machine without depending on external S3 endpoint
